ImperatorToCK3/Source/Imperator/Families/Families.cpp

```diff
@@ -21,21 +20,19 @@ void Imperator::Families::loadFamilies(std::istream& theStream)
 
 void Imperator::Families::registerKeys()
 {
-	registerRegex("\\d+", [this](const std::string& theFamilyID, std::istream& theStream) {
+	registerMatcher(commonItems::integerMatch, [this](const std::string& theFamilyID, std::istream& theStream) {
 		const auto familyStr = commonItems::stringOfItem(theStream).getString();
-		if (familyStr.find('{') != std::string::npos)
-		{
+		if (familyStr.find('{') != std::string::npos) {
 			std::stringstream tempStream(familyStr);
-			if (families.contains(std::stoull(theFamilyID))) {
-				families[std::stoull(theFamilyID)]->updateFamily(tempStream);
-			}
-			else {
-				auto newFamily = std::make_shared<Family>(tempStream, std::stoull(theFamilyID));
-				families.insert(std::pair(newFamily->getID(), newFamily));
+			const auto ID = commonItems::stringToInteger<unsigned long long>(theFamilyID);
+			if (families.contains(ID)) {
+				Log(LogLevel::Debug) << "Redefinition of family " << theFamilyID;
 			}
+			std::shared_ptr<Family> newFamily = familyFactory.getFamily(tempStream, ID);
+			families.emplace(newFamily->getID(), newFamily);
```

Comment:
This will fail if an existing family has the same id: https://en.cppreference.com/w/cpp/container/map/emplace

If that's the intended behavior, make it clear with the if/else statements. If not, you can use this pattern to avoid an extra lookup in the map:
```
std::shared_ptr<Family> newFamily = familyFactory.getFamily(tempStream, ID);
auto [iterator, inserted] = families.emplace(newFamily->getID(), newFamily);
if (!inserted)
{
   Log(LogLevel::Debug) << "Redefinition of family " << theFamilyID;
    iterator.second = newFamily;
}
```

This might even be a good format for if you don't replace the family, but that may be a matter of taste.

Reply:
Thanks, used this pattern.